apps/pay/app/[username]/layout.tsx

```diff
@@ -1,19 +1,72 @@
 import React from "react"
 
+import Link from "next/link"
+
+import { ApolloError, ApolloQueryResult } from "@apollo/client"
+
+import { getClient } from "../ssr-client"
+
+import { defaultCurrencyMetadata } from "../currency-metadata"
+
+import styles from "./username.module.css"
+
 import UsernameLayoutContainer from "@/components/layouts/username-layout"
+import { InvoiceProvider } from "@/context/invoice-context"
+import {
+  AccountDefaultWalletsDocument,
+  AccountDefaultWalletsQuery,
+} from "@/lib/graphql/generated"
 
-export default function UsernameLayout({
-  children,
-  params,
-}: {
+type Props = {
   children: React.ReactNode
   params: {
     username: string
   }
-}) {
+}
+
+export default async function UsernameLayout({ children, params }: Props) {
+  let response: ApolloQueryResult<AccountDefaultWalletsQuery> | { errorMessage: string }
+  try {
+    response = await getClient().query<AccountDefaultWalletsQuery>({
+      query: AccountDefaultWalletsDocument,
+      variables: { username: params.username },
+    })
+  } catch (e) {
+    console.error("error in username-layout.tsx", e)
+    const error = e as ApolloError
```

Comment:
if we know this is an ApolloError, then the casting can be done in the catch branch. but generally it's better not to assume a specific type of error and using `if (err instanceof ApolloError)`